src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import { getCommonProps } from '../../utils';
+import { px } from '../../utils';
+
+import Select from '../Select/Select';
+import Button from '../Button/Button';
+import chevronRight from '../../assets/chevronRight.svg';
+
+export interface PaginationProps {
+  /**
+   * Unique id for component testing
+   */
+  id: string;
+  /**
+   * Base class for Pagination component.
+   */
+  className?: string;
+  /**
+   * Child element pass in to display as options to select from.
+   */
+  children?: React.ReactNode;
+  /**
+   * Boolean to specify whether the `<Pagination>` should be disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Optional text to display when disable is true
+   */
+  disabledText?: string;
+  /**
+   * Optional `onClick` handler that is called whenever the `<Pagination>` value is clicked
+   */
+  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  /**
+   * Optional `onChange` handler that is called whenever the `<Pagination>` value is changed
+   */
+  onChange?: (selectedValue: string) => void;
+  /**
+   * Specify the value of the `<Pagination>`
+   */
+  value?: string | undefined;
```

Comment:
does a `string` or a `number` make more sense?  To me `number` makes the most sense.  I also wonder should this be called `defaultValue` since after the component is rendered once it ignores this `value` prop and instead uses it's internal `setState` (i.e. if it changes it doesn't get updated